src/app/dev/platforms/desktop/DevToys.Linux/Core/ThemeListener.cs

```diff
@@ -147,8 +155,48 @@ private void UpdateSystemSettingsAndApplyTheme()
 
     private AvailableApplicationTheme GetCurrentSystemTheme()
     {
-        return _gtkSettings.GtkApplicationPreferDarkTheme || (_gtkSettings.GtkThemeName?.Contains("Dark", StringComparison.OrdinalIgnoreCase) ?? false)
-            ? AvailableApplicationTheme.Dark
-            : AvailableApplicationTheme.Light;
+        try
+        {
+            var bus = DBusConnection.Get(BusType.Session);
+            using var parameters = Variant.NewTuple([
+                Variant.NewString("org.freedesktop.appearance"), Variant.NewString("color-scheme")
+            ]);
+
+            using Variant ret = bus.CallSync(
+                busName: "org.freedesktop.portal.Desktop",
+                objectPath: "/org/freedesktop/portal/desktop",
+                interfaceName: "org.freedesktop.portal.Settings",
+                methodName: "Read",
+                parameters: parameters,
+                replyType: VariantType.New("(v)"),
+                flags: DBusCallFlags.None,
+                timeoutMsec: 2000,
+                cancellable: null
+            );
+
+            uint userThemePreference = ret.GetChildValue(0).GetVariant().GetVariant().GetUint32();
```

Comment:
Somewhere here you should, if I read the documentation right, check for an G_IO_ERROR_CLOSED error, in which case the fallback should be called.

Reply:
Thanks ! (and sorry for the late reply) 
I will try to do this soon. :)

Reply:
If an error occurs, the `bus.CallSync()` call will throw an exception as gir.core interpretes the error argument by itself and raises an exception. As there is already a `try/catch` in place this should be fine.

So you probably only need to explicitly define in the switch expression which Theme DevToys should use if the user has no preference instead of calling the fallback.

Reply:
Hello,
So sorry for taking so long to get back to you. I couldn't work on DevToys for about 2 months as I was very busy with other things in my life.
I applied the last recommendation you made. I think we're good to go here :)